libs/blocks/bulk/utils.js

```diff
@@ -0,0 +1,267 @@
+const BULK_STORED_RESULTS = 'bulkStoredResults';
+const BULK_STORED_OPERATION = 'bulkStoredOperation';
+const UNSUPPORTED_SITE_STATUS = 'unsupported domain';
+const DUPLICATE_STATUS = 'duplicate';
+
+// eslint-disable-next-line no-promise-executor-return
+const delay = (milliseconds) => new Promise((resolve) => setTimeout(resolve, milliseconds));
+
+export const getStoredOperation = () => {
+  const name = getLocalStorage(BULK_STORED_OPERATION);
+  const urlIdx = getLocalStorage(BULK_STORED_URL_IDX);
+  const urls = getLocalStorage(BULK_STORED_URLS);
+  const completed = getLocalStorage(BULK_STORED_COMPLETED);
+  return {
+    name,
+    urlIdx,
+    urls,
+    completed,
+  };
+};
+
+export const getStoredUrlInput = () => {
+  const storedUrls = getStoredOperation().urls;
+  return storedUrls ? storedUrls.join('\n') : '';
+};
+
+export const storeUrls = (urls) => {
+  setLocalStorage(BULK_STORED_URLS, urls);
+};
+
+export const storeOperation = (operation) => {
+  setLocalStorage(BULK_STORED_OPERATION, operation);
+};
+export const getUser = async () => {
+  const profile = await window.adobeIMS?.getProfile();
+  return profile
+    ? { name: profile.name, email: profile.email }
+    : { name: 'anonymous', email: 'anonymous' };
+};
+
+export const signIn = async () => {
+  // force user to sign in
+  const accessToken = await getImsToken(loadScript);
+  if (!accessToken) {
+    window.adobeIMS.signIn();
+    return false;
+  }
+  return true;
+};
+
+export const signOut = (e) => {
+  e.preventDefault();
+  window.adobeIMS?.signOut();
+};
+
+export const getAuthorizedUsers = async () => {
+  let authorizedUsers = getLocalStorage(BULK_AUTHORIZED_USERS);
+  if (authorizedUsers) return authorizedUsers;
+  const resp = await fetch(BULK_CONFIG_FILE_PATH);
+  const json = await resp.json();
+  authorizedUsers = json.users.data.map((user) => user.user);
+  setLocalStorage(BULK_AUTHORIZED_USERS, authorizedUsers);
+  return authorizedUsers;
+};
+
+const getSupportedSites = async () => {
+  let supportedSites = getLocalStorage(BULK_SUPPORTED_SITES);
+  if (supportedSites) return supportedSites;
+  const resp = await fetch(BULK_CONFIG_FILE_PATH);
+  const json = await resp.json();
+  supportedSites = json.sites.data.map((site) => site.origin);
+  setLocalStorage(BULK_SUPPORTED_SITES, supportedSites);
+  return supportedSites;
+};
+
+export const userIsAuthorized = async () => {
+  const { email } = await getUser();
+  const users = await getAuthorizedUsers();
+  return users.includes(email);
+};
+
+const siteIsSupported = async (url) => {
+  try {
+    const { origin } = new URL(url);
+    const sites = await getSupportedSites();
+    return sites.includes(origin);
+  } catch (e) {
+    return false;
+  }
+};
+
+export const getUrls = (element) => {
+  return element.current?.value.split('\n')
+    .filter((url) => url.length > 0)
+    .map((e) => e.trim());
+};
+
+export const getActionName = (action, gerund) => {
```

Comment:
What is `gerund`?

Reply:
@chrischrischris it's a flag to either return the verb or its present participe form (root verb + 'ing'): 
getActionName('preview') returns 'Preview'
getActionName('preview', true) returns 'Previewing'
...

Reply:
to make it clearer I could rename it `useGerund`

Reply:
Ah ok - I learned a new word today!